qiskit_aer/backends/aer_compiler.py

```diff
@@ -465,7 +465,13 @@ def assemble_circuit(circuit: QuantumCircuit):
 
     qreg_sizes = []
     creg_sizes = []
-    global_phase = float(circuit.global_phase)
+    if (
+        isinstance(circuit.global_phase, ParameterExpression)
+        and len(circuit.global_phase.parameters) > 0
+    ):
+        global_phase = 1.0
```

Comment:
Very unimportant, but I'd think that the more natural default is 0, not 1 - `QuantumCircuit.global_phase` is the angle, not the unitary value.